salt/states/consul.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+'''
+Consul Managemnt
+===========================
+
+The consul module is used to create and manage ACL
+
+.. code-block:: yaml
+
+    acl_present:
+      consul.acl_present:
+        - id: 38AC8470-4A83-4140-8DFD-F924CD32917F
+        - name: acl_name
+        - rules: node "" {policy = "write"} service "" {policy = "read"} key "_rexec" {policy = "write"}
+        - type: client
+        - consul_url: http://localhost:8500
+
+    acl_delete:
+       consul.acl_absent:
+         - id: 38AC8470-4A83-4140-8DFD-F924CD32917F
+'''
+
+
+def _acl_changes(name, id=None, type=None, rules=None, consul_url=None, token=None):
+    '''
+       return True if the acl need to be update, False if it doesn't need to be update
+    '''
+    r = __salt__['consul.acl_info'](id=id, token=token, consul_url=consul_url)
+
+    if r['res'] and r['data'][0]['Name'] != name:
+        return True
+    elif r['res'] and r['data'][0]['Rules'] != rules:
+        return True
+    elif r['res'] and r['data'][0]['Type'] != type:
+        return True
+    else:
+        return False
+
+
+def _acl_exists(name=None, id=None, token=None, consul_url=None):
+    '''
+       return True if acl exist, if False it not
+    '''
+
+    res = {'result': False, 'id': None}
+
+    if id:
+        r = __salt__['consul.acl_info'](id=id, token=token, consul_url=consul_url)
+    elif name:
+        r = __salt__['consul.acl_list'](token=token, consul_url=consul_url)
+
+    if len(r['data']):
```

Comment:
Can `if r.get('data')` work here instead of `len`? That might be a little more clear here.